src/core/processing/qgsprocessingparameters.cpp

```diff
@@ -695,7 +701,7 @@ QString QgsProcessingParameters::parameterAsFileOutput( const QgsProcessingParam
   if ( dest == QgsProcessing::TEMPORARY_OUTPUT )
   {
     if ( const QgsProcessingDestinationParameter *destParam = dynamic_cast< const QgsProcessingDestinationParameter * >( definition ) )
-      val = destParam->generateTemporaryDestination();
```

Comment:
@nyalldawson not 100% sure, but seems there is a typo here and temporary file was never resolved to the correct path. Fixed it too, when adding test for temporary directory output.